internal/driver/onvifclient.go

```diff
@@ -343,7 +346,10 @@ func (onvifClient *OnvifClient) callGetSnapshotFunction() ([]byte, errors.EdgeX)
 	if edgexErr != nil {
 		return nil, errors.NewCommonEdgeXWrapper(edgexErr)
 	}
-	uriResponse := respContent.(*media.GetSnapshotUriResponse)
+	uriResponse, ok := respContent.(*media.GetSnapshotUriResponse)
+	if !ok {
+		return nil, errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("invalid GetSnapshotUriResponse for the camera %s", onvifClient.DeviceName), nil)
```

Comment:
I would probably add the type of the response object to aid with debugging. _Rinse and repeat for all similar calls._
```suggestion
		return nil, errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("invalid GetSnapshotUriResponse of type %T for the camera %s", uriResponse, onvifClient.DeviceName), nil)
```

Reply:
Added.